### ISSUE TITLE
BT-67(a)-Procedure: Exclusion Grounds

### ISSUE THREAD

**Issue by duncandewhurst** [open-contracting/european-union-support#173]
Follow up to a [Slack discussion](https://app.slack.com/client/T01GNNR4Y94/C03S8JL100P/thread/C03S8JL100P-1661872262.757289)

@jpmckinney, I'm drafting the exclusion grounds extension based on the modelling used in the [selection criteria extension](https://extensions.open-contracting.org/en/extensions/selectionCriteria/master/), which references the [selectionCriteriaType.csv](https://extensions.open-contracting.org/en/extensions/selectionCriteria/master/codelists/#selectionCriterionType.csv) codelist in `SelectionCriterion.type`.

In eForms, [BT-67(a)-Procedure](https://docs.ted.europa.eu/eforms/latest/schema/procedure-lot-part-information.html#exclusionGrounds) references the [exclusion-ground codelist](https://docs.ted.europa.eu/eforms/latest/reference/code-lists/exclusion-ground.html#_codes). Rather than mapping BT-67(a)-Procedure to an equivalent `ExclusionCriteria.type` field, I'm thinking a better approach would be to add an `ExclusionCriteria.classification` field and reference the eForms codelist using the `Classification` object. That way, `.type` is available if we decide to add a standardized codelist in the future. Does that sound good?

**Comment by jpmckinney** [comment]
Hmm, I think here it's fine to just have a single field (`type`), and the publication policy/user guide says where the codes are from.

There aren't many formal classifications for exclusion grounds, which is where a classification object starts to look appealing.

We can leave the door open to adding our own codelist in the future, but it'll be an open codelist anyhow.

In retrospect, `procurementMethodRationaleClassifications` and `unstructuredChanges` were maybe over-engineered in their use of Classification - but at least they are limited to the EU profile(s).

**Comment by duncandewhurst** [comment]
Sounds good!

**Comment by duncandewhurst** [comment]
@jpmckinney, I've updated the [exclusion grounds extension](https://github.com/open-contracting-extensions/ocds_exclusionGrounds_extension) and `guidance.yaml` (in 8932a1d706e06e670170e395aace8d9e1acd58ee). Please could you review it?

Let me know if there's a better way of refactoring extensions than forking and committing to `main` in the fork.

**Comment by duncandewhurst** [comment]
Follow-up question: if using the single `.type` field for the code, should we also map the code's description somewhere?

**Comment by jpmckinney** [comment]
guidance.yaml looks good. What is test.json in the extension? We can order `type` before `description` in the schema. Otherwise the extension looks good.

I suppose since we have a description field available we could use it, like we do for Classification.description.

Note that extensions should use `master`, to be consistent with old extensions. I renamed the branch.

Instead of creating a fork, I would just create a new repository from scratch. Unfortunately you can't convert a fork to a normal repository without contacting GitHub Support. If you don't have the privileges to recreate the repo, I'll do it.

**Comment by duncandewhurst** [comment]
I've created a [new repository](https://github.com/open-contracting-extensions/ocds_exclusionGrounds_extension) and renamed and archived the [old fork](https://github.com/open-contracting-extensions/ocds_exclusionGrounds_extension_old). I don't have permission to delete it.

`test.json` was committed in error, I've removed it from the new repository.

`tender.exclusionGrounds.criteria` is already mapped in `BT-67(b)-Procedure`, which is a free-text description of the exclusion criteria. I think it's fine to concatenate the code's description to that field. Sound good?
